modules/powers/Nemesis.class.php

```diff
@@ -15,7 +15,148 @@ public function __construct($game, $playerId)
     $this->playerCount = [2, 3, 4];
     $this->golden  = false;
     $this->orderAid = 42;
+
+    $this->implemented = true;
+  }
+
+
+  public function stateAfterBuild()
+  {
+    $arg = [];
+    $this->argUsePower($arg);
+    return (count($arg['workers']) > 0)? 'power' : null;
+  }
+
+
+  public function argUsePower(&$arg)
+  {
+    $arg['power'] = $this->id;
+    $arg['power_name'] = $this->name;
+    $arg['skippable'] = true;
+    $arg['workers'] = $this->game->board->getPlacedActiveWorkers();
+    foreach($arg['workers'] as &$worker)
+    {
+      $worker['works'] = [];
+    }
+	$workers = $this->game->board->getPlacedActiveWorkers();
+	if (count($workers) != 2)
+	  throw new BgaVisibleSystemException('Unexpected state in Nemesis.');
+
+	$actions = $this->game->log->getLastWorks(['force'], null, 2);
+
+	# choose second exchange
+	if (count($actions) > 0)
+	{		
+	    $arg['skippable'] = false;
+		if (count($actions) != 2)
+		    throw new BgaVisibleSystemException('Unexpected log in Nemesis.');
+		    
+		$lastoppw = $actions[0]['pieceId'];
+		$lastw = $actions[1]['pieceId'];
+	
+		$opponent = $this->game->board->getPiece($lastoppw)['player_id'];	
+	    $oppWorkers = $this->game->board->getPlacedWorkers($opponent);
+	    
+	    Utils::filterWorkersById($oppWorkers, $lastoppw, false);
+	    Utils::filterWorkersById($workers, $lastw, false);
+	    
+	    
+		if (count($workers) != 1)
+		    throw new BgaVisibleSystemException('Unexpected state in Nemesis.');
+	    
+	    $arg['workers'] = $workers;
+	    $arg['workers'][0]['works'] = $oppWorkers;
+	
+	}
+	else
+	{
+		$opponents = $this->game->playerManager->getOpponentsIds();
+
+		
+
+		// for each opponent, check if no worker neighbors Nemesis'    
+		foreach($opponents as $opp){
+		  $oppWorkers = $this->game->board->getPlacedWorkers($opp);
```

Comment:
I believe this case should not happen as Nemesis is banned vs Clio for this reason: she should always see at least two workers from each opponent so that she has the possibility to use her power

Reply:
Safer to use the function we have for this purpose, just in case...

Reply:
using this would need to filter again opponent by opponent to handle 3-players game I think. Tried a bit and could not work it out so I just left it as I think Nemesis power bans all such cases.